### PR TITLE
Merge Zach's many changes into master

### PR DESCRIPTION
**Pull request by fztgreen**
Merging
Rotation
Levels
XML Changes (more positional variety of platforms, stolen object)
Framerate management
Other changes etc.

### CODE REVIEW

File: include/Level.h [hunk 1]
```diff
@@ -19,18 +19,26 @@ namespace escape{
       int backgroundStartX;
       int backgroundStartY;
 
+      int startX;
```
Comment (wulind) [inline include/Level.h]: remove startX & startY; these are backgroundStartX & backgroundStartY

File: src/EventManager.cpp [hunk 2]
```diff
@@ -29,6 +29,10 @@ void EventManager::updateMouse(sf::Vector2i mousePosition, std::vector<Platform>
 void EventManager::checkMouseOverPlatform(sf::Vector2i mousePosition, std::vector<Platform> &platforms){
 	int i = 0;
 	for (i; i < platforms.size(); ++i){
+		//If platform is not draggable, ignore
+		if (platforms[i].draggable == false){
```
Comment (wulind) [inline src/EventManager.cpp]: Put not draggable platforms in a different array

File: src/GameLogic.cpp [hunk 3]
```diff
@@ -9,10 +9,16 @@ GameLogic::GameLogic(){
 	//Game State
 	this -> state.setState(GameState::State::TITLE);
 
-	//Initializes world.
-	//Takes in Gravity (change second param to change gravity)
-	b2Vec2 Gravity(0.f, 15.f);
-	this -> World = new b2World(Gravity);
+	//LevelFactory & Level
+	this -> factory = LevelFactory();
```
Comment (wulind) [inline src/GameLogic.cpp]: remove

File: src/GameLogic.cpp [hunk 4]
```diff
@@ -9,10 +9,16 @@ GameLogic::GameLogic(){
...
+
+	//TitlePage
+	this -> titlePage = TitlePage();
```
Comment (wulind) [inline src/GameLogic.cpp]: remove

File: src/GameLogic.cpp [hunk 5]
```diff
@@ -9,10 +9,16 @@ GameLogic::GameLogic(){
...
+
+	//Resources
+	this -> resources = ResourceManager();
```
Comment (wulind) [inline src/GameLogic.cpp]: remove

File: src/Level.cpp [hunk 6]
```diff
@@ -4,8 +4,7 @@ using namespace escape;
 Level::Level(){
   // this -> platforms.push_back(Platform(0, 300, 10, 200, this -> World));//TODO: move into respective places
   this -> base = Platform(0, 600, 10, 800);
-  this -> stolenObject = StolenObject(50, 100, 32);
-
+  //this -> stolenObject = StolenObject(50, 100, 32);
```
Comment (wulind) [inline src/Level.cpp]: remove commented code

File: src/Level.cpp [hunk 7]
```diff
@@ -4,8 +4,7 @@ using namespace escape;
 Level::Level(){
   // this -> platforms.push_back(Platform(0, 300, 10, 200, this -> World));//TODO: move into respective places
```
Comment (wulind) [inline src/Level.cpp]: remove commented code

File: src/LevelFactory.cpp [hunk 8]
```diff
@@ -56,9 +58,23 @@ void LevelFactory::makePlatforms(Level &level, tinyxml2::XMLElement *levelRoot){
   tinyxml2::XMLElement *child = levelRoot -> FirstChildElement("Platforms") -> FirstChildElement("Platform");
 
   while(child != nullptr){
-    level.makePlatform(std::atoi(child -> Attribute("rotation")), yPos);\
+    bool draggable;
+    if (std::atoi(child -> Attribute("draggable")) == 1){
```
Comment (wulind) [inline src/LevelFactory.cpp]: get rid of "==1"

File: src/Platform.cpp [hunk 9]
```diff
@@ -72,6 +77,8 @@ void Platform::updateDragPosition(float mouseX, float mouseY) {
 		this -> xCoord = mouseX - mouseDragOffsetX;
 		this -> yCoord = mouseY - mouseDragOffsetY;
 		this -> Body -> SetTransform(b2Vec2(this -> xCoord / SCALE, this -> yCoord / SCALE), this -> rotation * b2_pi / 180);
+
```
Comment (wulind) [inline src/Platform.cpp]: remove extra lines

File: src/EventManager.cpp (hunk 2 above)
Comment (fztgreen) [reply src/EventManager.cpp]: Not happening on the fly in this pull request. Noted for later.

**Pull request merged**